src/components/input/ExpressionInput.tsx

```diff
@@ -21,6 +21,8 @@ type Props = {
   titleTooltip?: string;
   /** Maximum width of the number input, in monospace characters */
   maxWidthCharacters?: number;
+  /** A function to verify the validity of an input. */
+  valid: ((thing: IExpressionStore) => boolean) | undefined;
```

Comment:
```suggestion
  valid?: ((thing: IExpressionStore) => boolean);
```
As far as TS is concerned, you want `?` for something that doesn't need to be written, but `| undefined` if it must be written but could be `valid={undefined}` (kinda pointless here but more generally useful.

This is the issue giving CI errors.

Reply:
I was unclear about that, thanks for clarifying